### PR TITLE
Add puppeteer video recorder helper

### PR DESCRIPTION
**Pull request by mstrofbass**
Proposal to add a Puppeteer video recording helper to make it easier to debug puppeteer tests.

I'm going to be using this regardless of whether it's merged or not; I just figured I'd offer it up since it may prove to be useful for others.

### CODE REVIEW

**Comment by mstrofbass** [comment]
This has puppeteer-screen-recorder pinned to version 3.0.3 because 3.05 seems to have a bug with the build/packaging that breaks importing. It should be unpinned once that is resolved in the dependency.

File: src/e2e/puppeteer/helpers/recordVideo.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { PuppeteerScreenRecorder } from 'puppeteer-screen-recorder'
+import { page } from '../setup'
+
+/** Initialize recorder. */
+async function init() {
```
Comment (raineorshine) [inline src/e2e/puppeteer/helpers/recordVideo.ts:5]: `init` should not be async

**Review by raineorshine — CHANGES_REQUESTED** [review]
Sure! This looks useful.

File: src/e2e/puppeteer/helpers/recordVideo.ts (hunk 1 above)
Comment (mstrofbass) [reply src/e2e/puppeteer/helpers/recordVideo.ts]: Good catch!

**Comment by mstrofbass** [comment]
When I start using it a little more I'll look into configuring a directory and then adding that directory to the .gitignore. We could probably just add `*.mp4` to the .gitignore file for now as well, but I should have a more robust solution soon enough.

**Comment by raineorshine** [comment]
FYI I decided to remove puppeteer-screen-recorder because the ffmpeg dependency was a bit heavy and it's not being actively used by the team.

**Pull request merged**